util/simple_mixed_compressor.cc

```diff
@@ -0,0 +1,101 @@
+#include "simple_mixed_compressor.h"
+
+#include <options/options_helper.h>
+
+#include <random>
+
+#include "rocksdb/advanced_compression.h"
+namespace ROCKSDB_NAMESPACE {
+
+// MultiCompressorWrapper implementation
+MultiCompressorWrapper::MultiCompressorWrapper(const CompressionOptions& opts,
+                                               CompressionType type,
+                                               CompressionDict&& dict) {
+  assert(type != kNoCompression);
+  assert(type == kZSTD);
+  auto builtInManager = GetDefaultBuiltinCompressionManager();
+  const auto& compressions = GetSupportedCompressions();
+  for (auto type_ : compressions) {
+    if (type_ == kNoCompression) {
+      continue;
+    }
+    compressors_.push_back(builtInManager->GetCompressor(opts, type_));
+  }
+  (void)dict;
+  (void)type;
+}
+
+size_t MultiCompressorWrapper::GetMaxSampleSizeIfWantDict(
+    CacheEntryRole block_type) const {
+  return compressors_.back()->GetMaxSampleSizeIfWantDict(block_type);
+}
+
+Slice MultiCompressorWrapper::GetSerializedDict() const {
+  return compressors_.back()->GetSerializedDict();
+}
+
+CompressionType MultiCompressorWrapper::GetPreferredCompressionType() const {
+  return kZSTD;
+}
+
+Compressor::ManagedWorkingArea MultiCompressorWrapper::ObtainWorkingArea() {
+  return compressors_.back()->ObtainWorkingArea();
+}
+
+std::unique_ptr<Compressor> MultiCompressorWrapper::MaybeCloneSpecialized(
+    CacheEntryRole block_type, DictSampleArgs&& dict_samples) {
+  return compressors_.back()->MaybeCloneSpecialized(block_type,
+                                                    std::move(dict_samples));
+}
+
+// SimpleMixedCompressor implementation
+Status SimpleMixedCompressor::CompressBlock(
+    Slice uncompressed_data, std::string* compressed_output,
+    CompressionType* out_compression_type, ManagedWorkingArea* wa) {
+  std::random_device rd;
```

Comment:
We have a preference for using our own Random library, starting with Random::GetTLSInstance(). (We don't need Mersenne Twister quality here.)